packages/ra-core/src/core/CoreAdminContext.tsx

```diff
@@ -36,6 +36,9 @@ export interface AdminContextProps {
     dashboard?: DashboardComponent;
     dataProvider?: DataProvider | LegacyDataProvider;
     queryClient?: QueryClient;
+    /**
+     * @deprecated Wrap your Admin inside a Router to change the routing strategy
+     */
```

Comment:
Why ? Shouldn't we add this warning in v3 and remove this prop completely here?

Reply:
This is deprecated because the only reason you may want to pass a custom history was to use another router. Now you can use anoyther router.

As for preparing the v4 migration in v3, it's a general matter that doesn't concern this PR. Besides, we couldn't deprecate it in v3 without supporting the new way, which isn't possible in v3 which uses the old react router.